### PR TITLE
Add slash ended urls for service-account-issuer-discovery to match API in swagger

### LINKED ISSUE #117455: Mismatched URLs for service-account-issuer-discovery and Open API Spec's swagger.json
### What happened?

When I use the official Kubernetes Java Client (version 18.0.0) from within a pod using `Config.defaultClient()` to build an `ApiClient` and then access `getServiceAccountIssuerOpenIDConfiguration` in the `WellKnownApi` I get the following error:

```json
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "forbidden: User \"system:serviceaccount:test:user\" cannot get path \"/.well-known/openid-configuration/\"",
  "reason": "Forbidden",
  "details": {},
  "code": 403
}
```

The same happens with the `OpenidApi#getServiceAccountIssuerOpenIDKeyset`.

However, the problem does not happen when I call the `/.well-known/openid-configuration` endpoint using the pod's service account and curl.

The problem is that the generated Open API spec indicates that the client should call `/.well-known/openid-configuration/` instead of `/.well-known/openid-configuration` (and similarly `/openid/v1/jwks/` instead of `/openid/v1/jwks`). While the API Server seems to resolve these endpoints just fine, the `ClusterRole` defined in the code does not have the trailing slashes and therefore rejects the requests with a 403:

https://github.com/kubernetes/kubernetes/blob/6a75e7c40cab5a0d96d815c3aa976dc98b4b0972/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L521-L530

### What did you expect to happen?

I expected to be able to access these APIs from within a pod that has a mounted service account because this [documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-issuer-discovery) suggests all that is required is a service account and because all clusters get this `ClusterRoleBinding` and this `ClusterRole` by default:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  labels:
    kubernetes.io/bootstrapping: rbac-defaults
  name: system:service-account-issuer-discovery
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:service-account-issuer-discovery
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:serviceaccounts
```

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  labels:
    kubernetes.io/bootstrapping: rbac-defaults
  name: system:service-account-issuer-discovery
rules:
- nonResourceURLs:
  - /.well-known/openid-configuration
  - /openid/v1/jwks
  verbs:
  - get
```

### How can we reproduce it (as minimally and precisely as possible)?

Here is the Java snippet to reproduce it:

```Java
        ApiClient k8sApiClient = Config.fromCluster();
        WellKnownApi wellKnownApi = new WellKnownApi(k8sApiClient);
        try {
            wellKnownApi.getServiceAccountIssuerOpenIDConfiguration();
        } catch (ApiException e) {
            LOG.error("Message: " + e.getMessage() + " Response body: " + e.getResponseBody());
        }
```

When run from within a pod that has a service account mounted into it, the above code will fail.

You can also reproduce the error without the Java Client by running `curl --cacert /run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $(cat /run/secrets/kubernetes.io/serviceaccount/token)"  https://kubernetes.default.svc/.well-known/openid-configuration/` from within a pod.

I can resolve the issue by updating the `system:service-account-issuer-discovery` cluster role to have the endpoints with the trailing `/`.

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  labels:
    kubernetes.io/bootstrapping: rbac-defaults
  name: system:service-account-issuer-discovery
rules:
- nonResourceURLs:
  - /.well-known/openid-configuration/
  - /.well-known/openid-configuration
  - /openid/v1/jwks/
  - /openid/v1/jwks
  verbs:
  - get
```

### Anything else we need to know?

I am happy to provide a fix, if I can get the right pointers. The detail I'm unclear on is how to fix this in a backwards compatible way.

Also, this PR seems to have introduced the potentially problematic `swagger.json` entry: https://github.com/kubernetes/kubernetes/pull/93258.

### Kubernetes version

<details>

Here is one of the remote clusters I reproduced the issues with.

```console
$ kubectl version
 kubectl version
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.3", GitCommit:"434bfd82814af038ad94d62ebe59b133fcb50506", GitTreeState:"clean", BuildDate:"2022-10-12T10:47:25Z", GoVersion:"go1.19.2", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.2", GitCommit:"9d142434e3af351a628bffee3939e64c681afa4d", GitTreeState:"clean", BuildDate:"2022-01-19T17:29:16Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.23) exceeds the supported minor version skew of +/-1
```

</details>

### Cloud provider

I reproduced this bug in multiple cloud providers as well as in minikube. It does not appear to be cloud provider specific. 

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
The service-account-issuer-discovery URLs(`/.well-known/openid-configuration` and `/openid/v1/jwks`) in swagger.json are slash ended, but the `system:service-account-issuer-discovery` is only allowed to access non slash ended endpoints.

In this case, if users follow the swagger, the request will be failed and make user confused.

#### Which issue(s) this PR fixes:

Fixes #117455

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by enj** [comment]
/assign @liggitt 

I will let Jordan decide which of the two PRs (or both) that he wants for this per

https://github.com/kubernetes/kubernetes/pull/110861#issuecomment-1534797580

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go [hunk 1]
```diff
@@ -524,7 +524,9 @@ func ClusterRoles() []rbacv1.ClusterRole {
 		Rules: []rbacv1.PolicyRule{
 			rbacv1helpers.NewRule("get").URLs(
 				"/.well-known/openid-configuration",
+				"/.well-known/openid-configuration/",
```
Comment (enj) [inline plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:527]: If we do decide to update the RBAC rules, this should have a comment that describes why we need the `/` terminated endpoints here.

**Comment by zelenushechka** [comment]
Hello @liggitt :blush:! I would like to reach out and confirm if this issue is still on track for the 1.28 release. Could you please provide an update on its progress? Thank you!

**Comment by zelenushechka** [comment]
Hey. I wanted to let you know that Code Freeze is coming up on [01:00 UTC Wednesday 19th July 2023 / 18:00 PDT Tuesday 18th July 2023](https://everytimezone.com/s/72ee8496), Please make sure that this PR is submitted before the deadline. (By the way, @haoruan, merging is blocked) We are currently in Week 7, and the Code Freeze will be in effect by Week 9.
Do let me know if I can be of any help :)

**Comment by zelenushechka** [comment]
Week 9 update: Code Freeze approaching!

Hey! We are currently in Week 9, and Code Freeze is just over a week away. It will begin at 01:00 UTC on Wednesday, 19th July 2023 / 18:00 PDT on Tuesday, 18th July 2023. **Is this PR on track for the 1.28 release?** Let me know if you need any assistance. :)

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go (hunk 1 above)
Comment (haoruan) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:528]: Sorry for the late, comment added

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go [hunk 2]
```diff
@@ -519,12 +519,15 @@ func ClusterRoles() []rbacv1.ClusterRole {
 	}...)
 
 	// Add the cluster role for reading the ServiceAccountIssuerDiscovery endpoints
+	// Also add slash ended URLs to allow non-golang clients to access apis without error
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go:522]:
```suggestion
	// Also allow slash-ended URLs to allow clients generated from published openapi docs prior to fixing the trailing slash to work properly
```

**Pull request merged**